servant/src/Servant/API/ContentTypes.hs

```diff
@@ -72,11 +72,8 @@ module Servant.API.ContentTypes
     , canHandleAcceptH
     ) where
 
-#if !MIN_VERSION_base(4,8,0)
-import           Control.Applicative              ((*>), (<*))
-#endif
 import           Control.Arrow                    (left)
-import           Control.Monad
+import           Control.Monad.Compat             hiding (mapM)
```

Comment:
Similarly here, if you use `base-compat`, you want to use `*.Compat` modules consistently to prevent name collisions.